embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/EmbraceImpl.java

```diff
@@ -704,6 +705,9 @@ private void startImpl(@NonNull Context context,
             internalEmbraceLogger.logDeveloper("Embrace", "Sending startup moment");
             nonNullEventService.sendStartupMoment();
         }
+
+        // This should return immediately given that EmbraceSpansService initialization should be fished at this point
+        spansInitTask.get();
```

Comment:
I'd suggesting bounding this to something like a few seconds, then no-oping/logging an internal error if it exceeds this time? Technically it should happen very rarely but I'm wary of anything that has the theoretical potential to indefinitely hang the SDK

Reply:
Haha, yeah, I was going to add that last night but it go too late. Good eye! We don't want this to block the app from starting